libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -994,36 +1007,29 @@ void _IotMqtt_DecrementConnectionReferences( _mqttConnection_t * pMqttConnection
 IotMqttError_t IotMqtt_Init( void )
 {
     IotMqttError_t status = IOT_MQTT_SUCCESS;
+    uint32_t initializationStatus = Atomic_CompareAndSwap_u32( &_initCalled,
+                                                               MQTT_LIBRARY_INITIALIZED,
+                                                               MQTT_LIBRARY_UNINITIALIZED );
 
-    if( _initCalled == 0 )
+    if( initializationStatus != 0 )
     {
         /* Call any additional serializer initialization function if serializer
          * overrides are enabled. */
         #if IOT_MQTT_ENABLE_SERIALIZER_OVERRIDES == 1
             #ifdef _IotMqtt_InitSerializeAdditional
                 if( _IotMqtt_InitSerializeAdditional() == false )
                 {
+                    /* Log initialization status. */
+                    IotLogError( "Failed to initialize MQTT library serializer. " );
+
                     status = IOT_MQTT_INIT_FAILED;
                 }
                 else
                 {
                     EMPTY_ELSE_MARKER;
                 }
-            #endif
+            #endif /* ifdef _IotMqtt_InitSerializeAdditional */
         #endif /* if IOT_MQTT_ENABLE_SERIALIZER_OVERRIDES == 1 */
-
-        /* Log initialization status. */
-        if( status != IOT_MQTT_SUCCESS )
-        {
-            IotLogError( "Failed to initialize MQTT library serializer. " );
-        }
-        else
-        {
-            /* Set the flag that specifies initialization is complete. */
-            _initCalled = 1;
-
-            IotLogInfo( "MQTT library successfully initialized." );
```

Comment:
This function doesn't log success anymore

Reply:
Good catch, re-added.